articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+
+#### Hapi.js
+Hapi.js's GitHub repository has 12.7k stars, 205 contributors, 1.3k forks and 332 releases.
+According to [npm](https://www.npmjs.com/package/hapi), 1,470 packages depend on Hapi.js.
+#### Koa.js
+Koa.js's GitHub repository has 29.9k stars, 218 contributors and 2.8k forks.
+According to [npm](https://www.npmjs.com/package/koa), 4,678 packages depend on Koa.js.
+#### Express.js
+Express.js's GitHub repository has 50k stars, 205 contributors, 8.3k forks and 280 releases.
+According to [npm](https://www.npmjs.com/package/express), 45,743 packages depend on Express.js.
+
+### Creating a server
+In this section we will have a look at how the frameworks differ in creating a server. Unlike Hapi, both Koa and Express use a similar logic to create a server.
+
+In Express, you first import the express module, then, instantiate the module and store it in a variable. Then use its listen method to start a server at a certain port.
+Lets see a code example to illustrate this.
+
+```JavaScript
+const express = require('express')
+const app = express()
+
+app.listen(2400, () => {console.log('Server started at port 2400')})
+```
+As we mentioned earlier, Koa uses the same logic as express. We import the Koa module, instantiate and store it in a variable and then use its listen method to start the server.
+The code snippet below shows how we start a server using Koa.js.
+
+```JavaScript
+const koa = require('koa')
+const app = koa()
+
+app.listen(2400, () => {console.log('Server started at port 2400')})
+```
+
+Now lets look at how Hapi stands out. In Hapi, we import the hapi module, then, instead of instantiating the module, we use its server method to create a server and then use its connection method to connect the server to a specific port. Then we use the start method to start the server.
+Lets see a code example.
+
+```JavaScript
+const hapi = require('hapi')
+const server = new hapi.Server()
+
+server.connection({
+    host: 'localhost',
+    port: '2400'
+})
+
+server.start(error => {
+    if (error) throw error
+    else console.log('Server running at PORT 2400');
+})
+```
+
+### Framework advantages and disadvantages
+In this section we are going to highlight some of the advantages and disadvantages of each framework.
+
+#### Hapi.js
+##### Advantages
+1. Hapi enables one to build a scalable web application or API.
+2. Hapi uses plugins which helps in scaling.
+3. Hapi can also be used in conjunction with front-end frameworks like Vue and React to build single-page web applications.
+4. Hapi provides a deeper control over request handling through the use of context.
+5. Hapi provides input/output validation, routing and caching, hence making it a good choice for building REST APIs.
+6. Hapi provides good plugins such as Yar, Good, Catbox etc.
```

Comment:
Since one of the main advantages of Hapi are its plugins, then it would be great if you could provide a code example of one.

Reply:
I demonstrated adding plugins on my introduction to hapi.js framework that's why I added a link to the article. I preferred not to add that so as to avoid repetition